pepr/utilities.py

```diff
@@ -84,3 +84,57 @@ def plot_class_dist_histogram(attack_alias, class_data, save_path):
     fig.savefig(save_path + f"/{path}")
     plt.close(fig)
     return path
+
+
+def create_attack_pars_table(report_section, values, pars_descriptors):
+    """
+    Generate LaTex table from pars_descriptors with fancy parameter descriptions.
+
+    Parameters
+    ----------
+    report_section : pepr.report.ReportSection
+        ReportSection container object for the current attack.
+    values : dict
+        Dictionary with the values stored for the keys in `pars_descriptors`. If
+        `pars_descriptors` has an entry "max_iter", then `values` should also have an
+        entry "max_iter" but holding the argument value instead of a description.
+    pars_descriptors : dict
+        Dictionary of attack parameters and their description shown in the attack
+        report.
+
+    Returns
+    -------
+    int
+        Number of table rows added.
+    """
+    from pylatex import Command, Tabular
+
+    new_rows = len(pars_descriptors)
+
+    if new_rows > 0:
+        # Create table for the attack parameters.
+        new_rows = len(pars_descriptors)
+        with report_section.create(Tabular("|l|c|")) as tab_ap:
+            for key in pars_descriptors:
+                # Add table row per pars_descriptor entry
+                desc = pars_descriptors[key]
+                if key in values:
+                    if isinstance(values[key], float):
+                        value = str(round(values[key], 3))
+                    else:
+                        value = str(values[key])
+                    tab_ap.add_hline()
+                    tab_ap.add_row([desc, value])
+            tab_ap.add_hline()
+        report_section.append(Command("captionsetup", "labelformat=empty"))
+        report_section.append(
+            Command(
+                "captionof",
+                "table",
+                extra_arguments="Attack parameters",
+            )
+        )
+    else:
+        report_section.append("Attack has no configuration parameters.")
+
+    return new_rows
```

Comment:
I thin we should make this function and the one above (create_attack_pars_table) private. Otherwise they occur in the documentation and I don't think they should. Or probably we should move them to their own module, because they are not utility for the user? What do you think?

Reply:
They are helpers for the report generation, so maybe we should move them to the `report.py` module

Reply:
Moved to `report.py` in 809befa